core/textile/bucket_factory.go

```diff
@@ -452,3 +465,93 @@ func GetDefaultBucketSlug() string {
 func GetDefaultMirrorBucketSlug() string {
 	return defaultPersonalMirrorBucketSlug
 }
+
+// Attempts to restore buckets from a hub replication
+// Returns nil if there's nothing to restore or the restoration succeeded
+func (tc *textileClient) restoreBuckets(ctx context.Context) error {
+	bucketList, err := tc.GetModel().ListBuckets(ctx)
+	if err != nil {
+		return err
+	}
+
+	if len(bucketList) == 0 && tc.shouldForceRestore {
```

Comment:
@dmerrill6 

Is `tc.shouldForceRestore` check necessary here?

Reply:
It's here because I don't want to error out when creating a new account (as it has no buckets), only when restoring an existing account